File: app/src/main/java/com/mubaiwa/trust/umkhathi/ui/Current.java
package com.mubaiwa.trust.umkhathi.ui;

import com.mubaiwa.trust.umkhathi.R;
import com.mubaiwa.trust.umkhathi.Weather.Forecast;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.TimeZone;

/**
 * Created by trust on 2016/01/30.
 */
public class Current {

    private String mIcon;
    private long mTime;
    private double mTemperature;
    private double mHumidity;
    private double mPrecipitation;
    private String mSummary;
    private String mTimeZone;

    public String getTimeZone() {
        return mTimeZone;
    }

    public void setTimeZone(String timeZone) {
        mTimeZone = timeZone;
    }

    public String getIcon() {
        return mIcon;
    }

    public void setIcon(String icon) {
        mIcon = icon;
    }

    public int getIconId(){

        return Forecast.getIconId(mIcon);
    }


    public long getTime() {
        return mTime;
    }


    public String getFormattedTime(){
        SimpleDateFormat formatter = new SimpleDateFormat("h:mm a");
        formatter.setTimeZone(TimeZone.getTimeZone(getTimeZone()));
        Date dateTime = new Date(getTime() * 1000);
        String timeString = formatter.format(dateTime);

        return timeString;
    }


    public void setTime(long time) {
        mTime = time;
    }

    public int getTemperature() {
        return (int)Math.round(mTemperature);
    }

    public void setTemperature(double temperature) {
        mTemperature = temperature;
    }

    public double getHumidity() {
        return mHumidity;
    }

    public void setHumidity(double humidity) {
        mHumidity = humidity;
    }

    public int getPrecipitation() {
        double precipPrecentage = mPrecipitation * 100;
        return (int)Math.round(precipPrecentage);
    }

    public void setPrecipitation(double precipitation) {
        mPrecipitation = precipitation;
    }

    public String getSummary() {
        return mSummary;
    }

    public void setSummary(String summary) {
        mSummary = summary;
    }
}

File: app/src/main/java/com/mubaiwa/trust/umkhathi/ui/Current.kt
package com.mubaiwa.trust.umkhathi.ui

import com.mubaiwa.trust.umkhathi.Weather.Forecast
import java.text.SimpleDateFormat
import java.util.*

/**
 * Created by trust on 2016/01/30.
 */
class Current {
    var icon: String? = null
    var time: Long = 0
    private var mTemperature = 0.0
    var humidity = 0.0
    private var mPrecipitation = 0.0
    var summary: String? = null
    var timeZone: String? = null

    val iconId: Int
        get() = Forecast.getIconId(icon ?: "")

    val formattedTime: String
        get() {
            val formatter = SimpleDateFormat("h:mm a")
            formatter.timeZone = TimeZone.getTimeZone(timeZone)
            val dateTime = Date(time * 1000)
            return formatter.format(dateTime)
        }

    val temperature: Int
        get() = Math.round(mTemperature).toInt()

    fun setTemperature(temperature: Double) {
        mTemperature = temperature
    }

    val precipitation: Int
        get() {
            val precipPrecentage = mPrecipitation * 100
            return Math.round(precipPrecentage).toInt()
        }

    fun setPrecipitation(precipitation: Double) {
        mPrecipitation = precipitation
    }

}